src/transformers/modeling_rag.py

```diff
@@ -0,0 +1,1359 @@
+# coding=utf-8
+# Copyright 2020, The RAG Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""RAG model implementation."""
+
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+
+import torch
+
+from .configuration_rag import RagConfig
+from .configuration_utils import PretrainedConfig
+from .file_utils import add_start_docstrings_to_callable, replace_return_docstrings
+from .modeling_outputs import ModelOutput
+from .modeling_utils import PreTrainedModel
+from .retrieval_rag import RagRetriever
+from .utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "RagConfig"
+
+
+@dataclass
+class RetrievAugLMMarginOutput(ModelOutput):
+    """
+    Base class for retriever augmented marginalized models outputs.
```

Comment:
Beautiful docstring! But as said in the doc file, we need to actually add them to the docs so users can see it :-)